cluster/kube-upgrade.sh

```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
+
+# Copyright 2015 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
+source "${KUBE_ROOT}/cluster/kube-env.sh"
+source "${KUBE_ROOT}/cluster/${KUBERNETES_PROVIDER}/util.sh"
+
+function usage() {
+  echo "${0} [-m|-n <node id>] <version>"
+  echo "  Upgrades master(-m) or specified node(-n)."
+  echo "  If the version is not specified will try to use local binaries."
+}
+
+upgrade_master=false
+upgrade_node=false
+
+while getopts "mn:h" opt; do
```

Comment:
As an experimental script sitting in `gce`, I was happy to dump a `getopts` script in. But we're about to ship this top-level. Does this UX look okay to people? @roberthbailey? It's not the prettiest in the world. *shrug*.

It's not like `kube-up.sh` and friends even have options, though.